test/iced_advanced.coffee

```diff
@@ -0,0 +1,151 @@
+if require?
+  iced = icedlib = require 'iced-runtime'
+
+##----------------------------------------------------------------------
+
+  atest "rendezvous & windowing example", (cb) ->
+
+    slots = []
+    call = (i, cb) ->
+      slots[i] = 1
+      await setTimeout(defer(), 10*Math.random())
+      slots[i] |= 2
+      cb()
+
+    window = (n, window, cb) ->
+      rv = new iced.Rendezvous
+      nsent = 0
+      nrecv = 0
+      while nrecv < n
+        if nsent - nrecv < window and nsent < n
+          call nsent, rv.id(nsent).defer()
+          nsent++
+        else
+          await rv.wait defer(res)
+          slots[res] |= 4
+          nrecv++
+      cb()
+
+    await window 10, 3, defer()
+    res = true
+    for s in slots
+      res = false unless s == 7
+    cb(res, {})
+
+##----------------------------------------------------------------------
+
+  atest "pipeliner example", (cb) ->
+
+    slots = []
+    call = (i, cb) ->
+      slots[i] = 1
+      await setTimeout(defer(), 3*Math.random())
+      slots[i] |= 2
+      cb(4)
+
+    window = (n, window, cb) ->
+      tmp = {}
+      p = new icedlib.Pipeliner window, .01
+      for i in [0..n]
+        await p.waitInQueue defer()
+        call i, p.defer tmp[i]
```

Comment:
this is an example in which `defer` is used not strictly as  keyword.   we can change pipeliner so we don't need this feature.